pkg/controller/endpointslice/endpointslice_controller.go

```diff
@@ -392,26 +393,18 @@ func (c *Controller) ensureSetupManagedByAnnotation(service *v1.Service) error {
 		return err
 	}
 
-	for _, endpointSlice := range endpointSlices {
-		if _, ok := endpointSlice.Labels[discovery.LabelManagedBy]; !ok {
-			if endpointSlice.Labels == nil {
-				endpointSlice.Labels = make(map[string]string)
-			}
-
-			endpointSlice.Labels[discovery.LabelManagedBy] = controllerName
-			_, err = c.client.DiscoveryV1beta1().EndpointSlices(endpointSlice.Namespace).Update(endpointSlice)
+	for _, epSlice := range endpointSlices {
+		if _, ok := epSlice.Labels[discovery.LabelManagedBy]; !ok {
+			epSlicePatch := []byte(fmt.Sprintf(`{"metadata":{"labels":{"%s":"%s"}}}`, discovery.LabelManagedBy, controllerName))
```

Comment:
don't construct json using sprintf, xref https://github.com/kubernetes/kubernetes/pull/81158

can make this once outside the loop with:
```go
serviceLabelPatch, err := json.Marshal(map[string]interface{}{
  "metadata": map[string]interface{}{
    "labels": map[string]interface{}{
      discovery.LabelManagedBy: controllerName,
    },
  },
})
```